source/WindowScroller/utils/dimensions.js

```diff
@@ -26,5 +26,5 @@ export function getHeight (element) {
  */
 export function getPositionFromTop (element, container) {
   const containerElement = container === window ? document.documentElement : container
-  return element.getBoundingClientRect().top - containerElement.getBoundingClientRect().top
+  return element.getBoundingClientRect().top + getVerticalScroll(container) - containerElement.getBoundingClientRect().top
```

Comment:
This fixes a bug which can be reproduced in the demo, before this commit, by

1. Check the "use custom element for scrolling" box
2. Press Shift + Tab twice so that the "Hide header text" button is focused
3. Scroll down a ways
4. Press the space bar so as to press "Hide header text"
5. Scroll again

I also made sure that the back-button navigation edge case still works.